Update Classic McEliece suppression files

praveksharma opened the pull request:
<!-- Please give a brief explanation of the purpose of this pull request. -->
Updates suppression files to include potential leaks document in #1540.
Updates Classic McEliece advisories to include information about environment-specific constant time leaks.
Removes Classic-McEliece-6XX from weekly constant time test skip list. 

<!-- Please answer the following questions to help manage version and changes across projects. -->

* [No] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [No] Does this PR change the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in [oqs-provider](https://github.com/open-quantum-safe/oqs-provider), [OQS-OpenSSL](https://github.com/open-quantum-safe/openssl), [OQS-BoringSSL](https://github.com/open-quantum-safe/boringssl), and [OQS-OpenSSH](https://github.com/open-quantum-safe/openssh) will also need to be ready for review and merge by the time this is merged.)

<!-- Once your pull request is ready for review and passing continuous integration tests, please convert from a draft PR to a normal PR, and request a review from one of the OQS core team members. -->

praveksharma commented:
@bhess I was hoping you could help me work on this issue, since I can't reproduce these errors locally. Could you check if Valgrind still complains about potential constant time leaks? 

It might be a good idea to add `--error-limit=no` to valgrind's [options list](https://github.com/open-quantum-safe/liboqs/blob/0dafd4f02c5a8fc000502ae51005a9962b671860/tests/test_constant_time.py#L193) to ensure valgrind reports all errors.

baentsch reviewed (commented):
Thanks for keeping updating the suppression files for this algorithm family.

As this PR adds this (IMO very correct) advisory:

> Current implementation of the algorithm may not be constant-time

and looking at the length/extent of the suppression files and the above advisory I wonder whether it wouldn't be simpler to _completely_ exclude McEliece from constant time testing until someone states interest in giving McEliece the CT property (?) We have something similar [for BIKE](https://github.com/open-quantum-safe/liboqs/blob/main/tests/constant_time/kem/issues/bike_has_no_timing_protections) -- although I wonder whether this is (still) correct given there's no corresponding advisory for BIKE (???).

bhess commented:
> @bhess I was hoping you could help me work on this issue, since I can't reproduce these errors locally. Could you check if Valgrind still complains about potential constant time leaks?

I ran the tests and there is one error remaining. Seems to be all fine when adding one more rule.
```
{
   This implementation of Classic McEliece may not be constant time.
   Memcheck:Value8
   src:pk_gen.c:322
   # fun:PQCLEAN_MCELIECE348864_AVX2_pk_gen
   fun:PQCLEAN_MCELIECE348864_AVX2_crypto_kem_keypair
}
```

praveksharma commented:
> looking at the length/extent of the suppression files and the above advisory I wonder whether it wouldn't be simpler to _completely_ exclude McEliece from constant time testing until someone states interest in giving McEliece the CT property (?) 

> We have something similar [for BIKE](https://github.com/open-quantum-safe/liboqs/blob/main/tests/constant_time/kem/issues/bike_has_no_timing_protections) -- although I wonder whether this is (still) correct given there's no corresponding advisory for BIKE (???).

Looking through the [file history](https://github.com/open-quantum-safe/liboqs/commits/main/tests/constant_time/kem/issues/bike_has_no_timing_protections) for the BIKE suppression file, it looks like the advisories weren't updated when the suppression file was created. I've added a similar advisory for BIKE. 

I've updated the suppression file with the remaining error, we should be good to go.

bhess reviewed (commented):
The McEliece ct-run looks ok on my test machine. Thanks for the update!

However, I didn't notice any BIKE errors. There also seem to be no issues registered in issues.json. I think [this file](https://github.com/open-quantum-safe/liboqs/blob/main/tests/constant_time/kem/issues/bike_has_no_timing_protections) could be removed since ct-fixes to BIKE were made in https://github.com/open-quantum-safe/liboqs/pull/1400.

baentsch commented on .github/workflows/weekly.yml:
```diff
@@ -16,7 +16,7 @@ jobs:
             CMAKE_ARGS: -DOQS_OPT_TARGET=generic -DCMAKE_BUILD_TYPE=Debug -DOQS_ENABLE_TEST_CONSTANT_TIME=ON
             PYTEST_ARGS: --numprocesses=auto -k 'test_constant_time'
-            SKIP_ALGS: 'SPHINCS\+-SHA*,Classic-McEliece-6(.)*'
+            SKIP_ALGS: 'SPHINCS\+-SHA*, Classic-McEliece-(.)*, BIKE(.)*'
```
Skipping BIKE seems incorrect as per #1400.

baentsch commented on docs/algorithms/kem/bike.md:
```diff
@@ -11,6 +11,9 @@
   - https://bikesuite.org/files/v5.0/Reference_Implementation.2022.10.04.1.zip
 
+## Advisories
+This implementation of BIKE is not constant time.
```
Changing this only in the .MD and not the .YML will not survive the next execution of `copy_from_upstream.py`.

baentsch commented on docs/algorithms/kem/classic_mceliece.md:
```diff
@@ -14,6 +14,7 @@
 ## Advisories
 
 - Classic-McEliece-460896, Classic-McEliece-460896f, Classic-McEliece-6960119, and Classic-McEliece-6960119f parameter sets fail memory leak testing on x86-64 when building with ``clang`` using optimization level ``-O2`` and ``-O3``. Care is advised when using the algorithm at higher optimization levels, and any other compiler and architecture.
+- Current implementation of the algorithm may not be constant-time. Additionally, environment specific constant-time leaks may not be documented; please report potential constant-time leaks when found. 
```
Same here. This ought to be in the YML (too) from which the MD is generated.

baentsch reviewed (changes requested):
Beyond the single comment made I concur with @bhess that 

> [this file](https://github.com/open-quantum-safe/liboqs/blob/main/tests/constant_time/kem/issues/bike_has_no_timing_protections) could be removed since ct-fixes to BIKE were made in https://github.com/open-quantum-safe/liboqs/pull/1400.
